packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -163,43 +171,36 @@ export default async function ShadowSystem(world: World) {
         index++
       }
 
-      const group = getComponent(entity, GroupComponent)
+      const dropShadowComponent = getComponent(entity, DropShadowComponent)
 
       const transform = getComponent(entity, TransformComponent)
 
       const raycaster = new Raycaster()
       raycaster.firstHitOnly = true
       raycaster.set(transform.position, shadowDirection)
 
-      const intersects = raycaster.intersectObjects(sceneObjects)
-      if (!intersects.length || !intersects[0].face) {
+      const intersected = raycaster.intersectObjects(sceneObjects)[0]
+      if (!intersected || !intersected.face) {
         setDropShadowMatrix(defaultShadowMatrix)
         continue
       }
 
-      const sphere = new Sphere()
-      new Box3().setFromObject(group[0], false).getBoundingSphere(sphere)
       const distanceShrinkBias = 3
-      const sizeBias = 1.5
-      const finalSize = sphere.radius * Math.min(distanceShrinkBias / intersects[0].distance, 1) * sizeBias
+      const sizeBias = 1
+      const finalSize = dropShadowComponent.radius * Math.min(distanceShrinkBias / intersected.distance, 1) * sizeBias
 
-      let shadowMatrix = new Matrix4()
-      const shadowRotation = new Quaternion().setFromUnitVectors(intersects[0].face.normal, V_001)
+      const shadowRotation = new Quaternion().setFromUnitVectors(intersected.face.normal, V_001)
```

Comment:
any `new` math objects being created often should instead be created once in module scope and reused for all repetitions